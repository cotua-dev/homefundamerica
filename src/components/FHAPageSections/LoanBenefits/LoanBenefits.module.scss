@import '../../../styles/variables';

section#loanBenefitsSection {
    height: 40vh;
    display: flex;
    align-items: center;
    justify-content: center;

    @media screen and (max-width: 950px) {
        height: 100%;
        padding: 40px 10px;
    }

    div.contentWrapper {
        h2.loanBenefitsTitle {
            color: $primaryColor;
            margin: 0;
            font-size: 32px;
            text-align: center;

            @media screen and (max-width: 950px) {
                font-size: 24px;
            }
        }

        div.benefitsWrapper {
            display: flex;
            flex-direction: row;
            margin-top: 20px;

            @media screen and (max-width: 950px) {
                flex-direction: column;
            }

            div.benefits {
                &:last-child {
                    margin-left: 20px;
                }

                @media screen and (max-width: 950px) {
                    &:last-child {
                        margin-left: 0;
                    }
                }

                div.benefit {
                    margin: 10px 0;

                    strong,
                    .icon {
                        font-size: 20px;

                        @media screen and (max-width: 950px) {
                            font-size: 14px;
                        }
                    }

                    strong {
                        color: $primaryColor;
                    }

                    .icon {
                        color: $secondaryColor;
                        margin-right: 10px;
                    }
                }
            }
        }
    }
}
