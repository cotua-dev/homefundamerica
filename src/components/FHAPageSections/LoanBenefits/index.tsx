import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck } from '@fortawesome/free-solid-svg-icons';
import { Benefit } from './LoanBenefits.interfaces';
import * as styles from './LoanBenefits.module.scss';

export const LoanBenefits: FunctionComponent = (): ReactElement => {
    const benefitsSetOne: Benefit[] = [
        {
            key: 0,
            benefit: 'Down payments as low as 3.5%',
        },
        {
            key: 1,
            benefit: 'Lower credit scores are accepted',
        },
        {
            key: 2,
            benefit: 'Flexible qualification guidelines',
        },
    ];

    const benefitsSetTwo: Benefit[] = [
        {
            key: 3,
            benefit: 'Fixed-rate and adjustable-rate loan options',
        },
        {
            key: 4,
            benefit: 'Refinance up to 96.5% of current home value',
        },
        {
            key: 5,
            benefit: 'Cash out and debt consolidation financing available',
        },
    ];

    return (
        <section id={styles.loanBenefitsSection}>
            <div className={styles.contentWrapper}>
                <h2 className={styles.loanBenefitsTitle}>Loan Benefits</h2>
                <div className={styles.benefitsWrapper}>
                    <div className={styles.benefits}>
                        {benefitsSetOne.map((benefit: Benefit) => (
                            <div className={styles.benefit} key={benefit.key}>
                                <FontAwesomeIcon className={styles.icon} icon={faCheck}/>
                                <strong>{benefit.benefit}</strong>
                            </div>
                        ))}
                    </div>
                    <div className={styles.benefits}>
                        {benefitsSetTwo.map((benefit: Benefit) => (
                            <div className={styles.benefit} key={benefit.key}>
                                <FontAwesomeIcon className={styles.icon} icon={faCheck}/>
                                <strong>{benefit.benefit}</strong>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </section>
    );
}
