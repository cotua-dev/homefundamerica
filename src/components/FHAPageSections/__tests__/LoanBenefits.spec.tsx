import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { LoanBenefits } from '../LoanBenefits';

describe('LoanBenefits Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<LoanBenefits/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
