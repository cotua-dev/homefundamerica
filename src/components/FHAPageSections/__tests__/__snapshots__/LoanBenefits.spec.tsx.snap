// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanBenefits Component renders 1`] = `
<section>
  <div>
    <h2>
      Loan Benefits
    </h2>
    <div>
      <div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Down payments as low as 3.5%
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Lower credit scores are accepted
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Flexible qualification guidelines
          </strong>
        </div>
      </div>
      <div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Fixed-rate and adjustable-rate loan options
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Refinance up to 96.5% of current home value
          </strong>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
            />
          </svg>
          <strong>
            Cash out and debt consolidation financing available
          </strong>
        </div>
      </div>
    </div>
  </div>
</section>
`;
